lib/liquid/standardfilters.rb

```diff
@@ -10,7 +10,7 @@ module StandardFilters
       '"'.freeze => '&quot;'.freeze,
       "'".freeze => '&#39;'.freeze
     }
-    HTML_ESCAPE_ONCE_REGEXP = /["><']|&(?!([a-zA-Z]+|(#\d+));)/
+    HTML_ESCAPE_ONCE_REGEXP = /["><']|&(?!([a-zA-Z]+\d*|(#\d+)|(#[xX][\da-fA-F]+));)/
```

Comment:
> doesn't mean they should have the same bugs

I agree with that, but I'm not convinced that it's a bug. If it's just a behaviour that you would prefer, then I would say no, lets do what Rails does. If it's actually a bug, then I think it should be fixed in both Liquid and Rails (not necessarily by yourself).

@rafaelfranca, what do you think?